test/distributed/_sharded_tensor/ops/test_linear.py

```diff
@@ -65,27 +68,62 @@ def _run_sharded_linear(self, spec, input_size, linear_size, sharded_dim):
         )
         self.assertEqual(local_output, sharded_output)
 
+        # Generate expected for loss calculation.
+        expected_result = torch.randint(0, 2, (input_size[0],)).cuda(self.rank)
+
+        # Compute loss.
+        local_loss = expected_result - torch.sum(local_output, dim=1)
+        sharded_loss = expected_result - torch.sum(sharded_output, dim=1)
+        self.assertEqual(local_loss, sharded_loss)
+
+        # Run backward pass
+        local_loss.backward(gradient=local_loss.detach())
+        local_grad = local_linear.weight.grad
+        sharded_loss.backward(gradient=sharded_loss.detach())
+
+        # Verify that both weight and bias in the sharded linear has non-None grad.
+        sharded_weight = sharded_linear.weight.local_shards()[0].tensor
```

Comment:
maybe we should introduce a sharded_tensor.grad attribute as we discussed, but the `.grad` probably should be a ShardedTensor as well (which contains local gradients for local shards), this can be a follow up that will make this test cleaner.

Reply:
sure.

Reply:
Lets create a gh issue for ShardedTensor.grad to track this.